.github/actions/setup-greptimedb-cluster/action.yml

```diff
@@ -0,0 +1,93 @@
+name: Setup GreptimeDB cluster
+description: Deploy GreptimeDB cluster on Kubernetes
+inputs:
+  frontend-replica-count:
+    default: 1
+    description: "Frontend replica count"
+  datanode-replica-count:
+    default: 2
+    description: "Datanode replica count"
+  metasrv-replica-count:
+    default: 1
+    description: "Metasrv replica count"
+  etcd-replica-count:
+    default: 1
+    description: "Etcd replica count"
+  image-registry: 
+    default: "docker.io"
+    description: "Image registry"
+  image-repository: 
+    default: "greptime/greptimedb"
+    description: "Image repository"
+  image-tag: 
+    default: "latest"
+    description: 'Image tag'
+  etcd-endpoints:
+    default: "etcd.etcd-cluster.svc.cluster.local:2379"
+    description: "Etcd endpoints"
+
+runs:
+  using: composite
+  steps:
+  - name: Install GreptimeDB operator
+    shell: bash
+    run: |
+      helm repo add greptime https://greptimeteam.github.io/helm-charts/ 
+      helm repo update
+      helm upgrade \
+        --install \
+        --create-namespace \
+        greptimedb-operator greptime/greptimedb-operator \
+        -n greptimedb-admin \
+        --wait \
+        --wait-for-jobs
+  - name: Install GreptimeDB cluster
+    shell: bash
+    run: | 
+      helm upgrade \
+        --install my-greptimedb \
+        --set meta.etcdEndpoints=${{ inputs.etcd-endpoints }} \
+        --set image.registry=${{ inputs.image-registry }} \
+        --set image.repository=${{ inputs.image-repository }}  \
+        --set image.tag=${{ inputs.image-tag }} \
+        greptime/greptimedb-cluster \
+        --create-namespace \
+        -n my-greptimedb \
+        --wait \
+        --wait-for-jobs
+  - name: Wait for GreptimeDB
```

Comment:
It's not a precise way to detect the readiness of the cluster. You should use the `.status.Phase` in `GreptimeDBCluster`.

For example:

```
function check_cluster_status() {
  wait_for \
    "kubectl get gtc <cluster-name> -n <namespace> -o json | jq '.status.clusterPhase' | grep Running" \
    "${DEFAULT_TIMEOUT%s}"
}

function wait_for() {
  local command="$1"
  local timeout="$2"
  local interval=1
  local elapsed_time=0

  while true; do
    # Turn off the exit-on-error flag to avoid exiting the script.
    set +e
    eval "$command" &> /dev/null
    local status=$?
    set -e

    # Check if command was successful.
    if [[ $status -eq 0 ]]; then
      return 0
    fi

    # Update elapsed time.
    ((elapsed_time += interval))

    # Check if the timeout has been reached.
    if [[ $elapsed_time -ge $timeout ]]; then
      echo "Timeout reached. Command failed."
      return 1
    fi

    # Wait for a specified interval before retrying.
    sleep $interval
  done
}
```

Reply:
I think the following code can do the same thing.
```bash
      kubectl wait \
        --for=condition=Ready \
        pod -l app.greptime.io/component=my-greptimedb-meta \
        --timeout=120s \
        -n my-greptimedb
      kubectl wait \
        --for=condition=Ready \
        pod -l app.greptime.io/component=my-greptimedb-datanode \
        --timeout=120s \
        -n my-greptimedb
      kubectl wait \
        --for=condition=Ready \
        pod -l app.greptime.io/component=my-greptimedb-frontend \
        --timeout=120s \
        -n my-greptimedb
```